src/FakeItEasy/Configuration/RuleBuilder.cs

```diff
@@ -158,9 +158,7 @@ public IAfterCallSpecifiedConfiguration CallsBaseMethod()
 
             public IReturnValueConfiguration<TMember> WhenArgumentsMatch(Func<ArgumentCollection, bool> argumentsPredicate)
             {
-                Guard.AgainstNull(argumentsPredicate, nameof(argumentsPredicate));

```

Comment:
So it does. Somehow I'd got myself looking at `AnyCallConfiguration.WhenArgumentsMatch `.
At least I was right about being wrong…